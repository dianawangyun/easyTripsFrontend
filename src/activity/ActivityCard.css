.list-group-item {
    background-color: rgba(100, 200, 250, 0.4);
}
