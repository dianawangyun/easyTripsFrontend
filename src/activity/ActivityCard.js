import React, { useState, useContext } from "react";
import EditActivityForm from "./EditActivityForm";
import TripContext from "../trip/TripContext";
import "./ActivityCard.css";

/* based on the props passed in render ActivityCard. */
function ActivityCard({
    activityId,
    activityName,
    location,
    latitude,
    longitude,
    description,
    startTime,
    endTime,
    comment,
    deleteActivity,
    isEditing,
    setIsEditing,
}) {
    const { setSelectLocation } = useContext(TripContext);
    /* if isUpdating, would render EditActivityForm for this activity. */
    const [isUpdating, setIsUpdating] = useState(false);
    const [deleteErrors, setDeleteErrors] = useState([]);

    async function handleDelete() {
        if (window.confirm("Are you sure to delete?")) {
            let res = await deleteActivity(activityId);
            if (!res.success) {
                setDeleteErrors(res.errs);
            }
        }
    }

    /* the function to setSelectLocation of this activity and interact with map. */
    function handleLocate() {
        if (latitude && longitude) {
            setSelectLocation({
                lat: parseFloat(latitude),
                lng: parseFloat(longitude),
            });
        }
    }

    const ShowActivityComp = (
        <li key={activityId} className="ml-3 mb-2 text-left list-group-item">
            <h5 className="card-title">{activityName}</h5>
            {description ? (
                <div>
                    <strong>Description: </strong>
                    {description}
                </div>
            ) : null}
            <div>
                <strong>Location: </strong>
                {location}
            </div>
            {startTime ? (
                <div>
                    <strong>Start Time: </strong>
                    {startTime}
                </div>
            ) : null}
            {endTime ? (
                <div>
                    <strong>End Time: </strong>
                    {endTime}
                </div>
            ) : null}
            {comment ? (
                <div>
                    <strong>Comment: </strong>
                    {comment}
                </div>
            ) : null}

            {isEditing ? null : (
                <div className="operations text-right">
                    {latitude && longitude ? (
                        <span className="mr-3" onClick={handleLocate}>
                            <i className="fas fa-search-location"></i>
                        </span>
                    ) : null}

                    <span
                        className="mr-3"
                        onClick={() => {
                            setIsEditing(true);
                            setIsUpdating(true);
                        }}
                    >
                        <i className="fas fa-edit"></i>
                    </span>
                    <span onClick={handleDelete}>
                        <i className="fas fa-trash-alt"></i>
                    </span>
                </div>
            )}

            {deleteErrors.length ? (
                <ul>
                    {deleteErrors.map((e) => (
                        <li className="text-danger">{e}</li>
                    ))}
                </ul>
            ) : null}
        </li>
    );

    return (
        <>
            {isUpdating ? (
                <EditActivityForm
                    activityId={activityId}
                    activityName={activityName}
                    location={location}
                    latitude={latitude}
                    longitude={longitude}
                    description={description}
                    startTime={startTime}
                    endTime={endTime}
                    comment={comment}
                    setIsEditing={setIsEditing}
                    setIsUpdating={setIsUpdating}
                />
            ) : (
                ShowActivityComp
            )}
        </>
    );
}
export default ActivityCard;
