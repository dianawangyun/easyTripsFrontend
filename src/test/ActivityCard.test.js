import React from "react";
import { render } from "@testing-library/react";
import ActivityCard from "../activity/ActivityCard";
import TripContext from "../trip/TripContext"


function renderActivityCard(setSelectLocation) {
    return render(<TripContext.Provider value={setSelectLocation}><ActivityCard />
        </TripContext.Provider>)
}

const setSelectLocation = () => {
    return;
}

it("renders without crashing", function () {
    renderActivityCard(setSelectLocation);
});

it("matches snapshot", function () {
    const { asFragment } = renderActivityCard(setSelectLocation);
    expect(asFragment()).toMatchSnapshot();
});

it("should show the activityCard", function () {
    const { queryByText } = renderActivityCard(setSelectLocation);
    expect(queryByText("Description:")).toBeInTheDocument();
});
