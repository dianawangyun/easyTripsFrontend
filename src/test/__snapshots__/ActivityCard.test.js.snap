// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="ml-3 mb-2 text-left list-group-item"
  >
    <h5
      class="card-title"
    />
    <div>
      Description: 
    </div>
    <div>
      Location: 
    </div>
    <div>
      Start Time: 
    </div>
    <div>
      End Time: 
    </div>
    <div>
      Comment: 
    </div>
    <div
      class="operations text-right"
    >
      <span
        class="mr-3"
      >
        <i
          class="fas fa-edit"
        />
      </span>
      <span>
        <i
          class="fas fa-trash-alt"
        />
      </span>
    </div>
  </li>
</DocumentFragment>
`;
